prediction_market_agent/agents/microchain_agent/nft_treasury_game/app_nft_treasury_game.py

```diff
@@ -185,6 +194,24 @@ def show_about_agent_part(nft_agent: type[DeployableAgentNFTGameAbstract]) -> No
         value=system_prompt,
         disabled=True,
     )
+    st.markdown("---")
+    with st.popover("Show unprocessed incoming messages"):
+        transactions = blockchain_transaction_fetcher().fetch_unseen_transactions(
+            nft_agent.wallet_address
+        )
```

Comment:
If I click 2x on the button, does it trigger `fetch_unseen_transactions` 2x?
`spice` is caching results, but still I would suggest placing the `transactions` inside a `st.session_state`, updating that through `st.fragment` and just reading from that inside the popover. Wdyt?

Reply:
This whole thing is in the fragment already:

![Screenshot by Dropbox Capture](https://github.com/user-attachments/assets/bf8815a4-1f8c-4637-9eed-7426a6522e50)

I don't see how saving it manually to state would help?

As for the button, I tested it and code inside of with block is executed regardless of clicking the button:

![Screenshot by Dropbox Capture](https://github.com/user-attachments/assets/4ab1b30f-0695-4502-9c26-b48f46b81a5c)

Reply:
Ow, understood - so button simply toggles the `disabled` property and shows stuff, but doesn't trigger any fetching logic. Fine.